### PR TITLE
Themes Tiers: Live preview aware of Personal Themes

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to:
- https://github.com/Automattic/dotcom-forge/issues/4597

## Proposed Changes
The scope of this PR is just to enable the Personal Tier for Theme Live Previewing, the texts and checkout URLs are out of scope and should not be considered.
In this PR, when you are on a Free Plan, Personal themes are now Live Previewable instead of being able to activate. Once you open the Live Preview you will see all the upsells and banners (they still reference Premium, the texts will be addressed on a separate PR)
* The Live Previewer is now aware of Personal Themes
* Once we release Theme Tiers we will need to refactor the Live Preview since it can be greatly simplified by using the theme_tier.feature prop instead of resolving the feature based on the type and then checking the active features.

## Testing Instructions

* Sandbox `widgets.wp.com`
* Open the Calyspo live link.
* Go to the LiTS and add `?flags=themes/tiers`.
* Filter by Personal Themes and click on the details of any Personal Theme.
* On the Theme Details page, click on the Live Preview button.
* You should see the same elements, upsells, and CTAs as if you were previewing a Personal theme. (Remember, the texts are not part of this PR for simplicity reasons)

Captura de pantalla 2023-12-21 a las 13 54 55
Captura de pantalla 2023-12-21 a las 13 55 24
Captura de pantalla 2023-12-21 a las 13 55 32

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
 	site?: SiteDetails;
 	previewingThemeType?: string;
+	requiredFeature?: string;
 } ) => {
 	const activeFeatures = site?.plan?.features?.active;
 	if ( ! activeFeatures ) {
 		return false;
 	}
 
+	// @TODO Replace all the logic below with the following code once Theme Tiers is live.
```
Comment (dsas) [inline apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts:32]: well, plus however the grandfathering logic works out 😆

File: apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts [hunk 2]
```diff
@@ -19,6 +20,25 @@ const getThemeType = ( theme?: Theme ) => {
...
+	if ( config.isEnabled( 'themes/tiers' ) ) {
+		return theme?.theme_tier?.slug ?? undefined;
+	}
+
+	return undefined;
+};
+
+/**
+ * Get the theme required feature.
+ * This only support WooCommerce and Premium themes.
```
Comment (dsas) [inline apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts:34]: this isn't true

Comment (rcrdortiz) [reply apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts:34]: Why you little nitpicker! Fixed, though, we're going to have to generalize or remove this description once we clean up the non-tier-related code.

**Review by dsas — APPROVED** [review]
Works as described in the scope for me 👍 

Worth adding the clean-up tasks for the list on https://github.com/Automattic/dotcom-forge/issues/4597

**Comment by rcrdortiz** [comment]
> Works as described in the scope for me 👍
> 
> Worth adding the clean-up tasks for the list on https://github.com/Automattic/dotcom-forge/issues/4597

Done, thanks for the quick review.

File: apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts [hunk 3]
```diff
@@ -58,10 +67,16 @@ export const useCanPreviewButNeedUpgrade = ( {
...
+
+		if ( config.isEnabled( 'themes/tiers' ) ) {
+			livePreviewUpgradeTypes.push( PERSONAL_THEME );
+		}
+
 		/**
 		 * Currently, Live Preview only supports upgrades for WooCommerce and Premium themes.
 		 */
-		if ( previewingTheme.type !== WOOCOMMERCE_THEME && previewingTheme.type !== PREMIUM_THEME ) {
+		if ( previewingTheme?.type && livePreviewUpgradeTypes.includes( previewingTheme.type ) ) {
```
Comment (okmttdhr) [inline apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts:79]: Hi @rcrdortiz! CC: @dsas 
This condition should be the opposite (`! livePreviewUpgradeTypes.includes`), as we need to show the upgrade notice for Premium/Woo themes.

I fixed it in https://github.com/Automattic/wp-calypso/pull/85771/.

**Pull request merged**